build_tools/ci/cpu_comparison/input_generator.py

```diff
@@ -0,0 +1,151 @@
+# This python script is expected to be run from the command-line with 2
+# arguments:
+#
+#   1) the name of a file to parse.
+#   2) the directory where binary files will be written.
+#
+# Example:
+# ```
+# python input_generator.py <input_file> <output_dir>
+# ```
+#
+# The file <input_file> contains an mlir function, and header information
+# about the inputs. See existing tests for examples.
+#
+# The header information about inputs is lines containing input types.
+# Example:
+#
+# ```
+#  # input 3x40xf32
+#  # input 2x2xi32
+# ```
+# This script finds all lines of the form above and generates binary files with
+# random data for them.
+#
+# This script also create a file containing a single line of the form
+# `--input="3x40xf32=@<binary_file>" --input="2x2xi32=@<binary_file>"`
+#
+# which will be used as input to iree-run-module in the main script.
+
+
+import numpy as np
+import struct
+import sys
+import os
+
+
+def float32_to_bfloat16(float32_value):
+
+    # Convert to a 32-bit integer representation
+    int32_repr = float32_value.view(np.int32)
+
+    # Truncate the lower 16 bits
+    bf16_int_repr = (int32_repr >> 16) & 0xFFFF
+
+    # Convert to numpy uint16 for consistency
+    return np.uint16(bf16_int_repr)
```

Comment:
I don't know where to get the reference. Is this the standard way to convert f32 to bf16? Also it may be better to rename this function as `convert_f32_to_bf16`.